src/coreComponents/constitutive/solid/PorousSolid.hpp

```diff
@@ -128,6 +145,50 @@ class PorousSolidUpdates : public CoupledSolidUpdates< SOLID_TYPE, BiotPorosity,
 //                                           volStrain );
   }
 
+  GEOSX_HOST_DEVICE
+  void smallStrainUpdateThermalSinglePhase( localIndex const k,
+                                            localIndex const q,
+                                            real64 const & initialFluidPressure,
+                                            real64 const & fluidPressure_n,
+                                            real64 const & fluidPressure,
+                                            real64 const & initialTemperature,
+                                            real64 const & temperature_n, 
+                                            real64 const & temperature,
+                                            real64 const ( &strainIncrement )[6],
+                                            real64 ( & totalStress )[6],
+                                            DiscretizationOps & stiffness ) const
+  {
+    real64 effectiveMeanStressIncrement = 0.0; 
+    
+    // Compute total stress increment and its derivative
+    computeTotalStressThermal( k,
+                               q,
+                               initialFluidPressure,
+                               fluidPressure,
+                               initialTemperature,
+                               temperature,
+                               strainIncrement,
+                               effectiveMeanStressIncrement, 
+                               totalStress,
+                               stiffness );
```

Comment:
> Maybe we should try to unify this function with the already existing `computeTotalStress` function. The unified function would always take pressure, temperature, and strain increment, and would always return the `effectiveMeanStressIncrement` and the derivatives (that, in your case, you would discard).
> 
> Then, we could pass a lambda that would be used to parameterize the function, for instance to compute the thermal-specific part. That would avoid some code duplication.

I have unified this function with the `computeTotalStress` function. I have moved the calculation of `effectiveMeanStressIncrement` out of this function, so that `computeTotalStress` can be the same for both fully coupled solver and fixed stress solver.